scripts/include/hook-common

```diff
@@ -0,0 +1,42 @@
+default_layouts="us"
+layouts_file="$SNAP/etc/layouts.txt"
+osk_layout_file="$SNAP_DATA/osk-layout.txt"
+
+# Use the first argument to join all the remaining ones
+function join {
+  local IFS="$1"
+  shift
+  echo "$*"
+}
+
+# Loop through list of layouts ensuring entries are valid and comma separated
+# Prints all the valid layouts and returns an error if any was not.
+function validate_layouts {
```

Comment:
I think this function is doing too many different things. There's two distinct tasks: filtering valid layouts and erroring if any layouts are invalid. I don't think combining them in a single function makes for clear code.